kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java

```diff
@@ -5,48 +5,36 @@
  */
 package io.kroxylicious.proxy.service;
 
-import io.kroxylicious.proxy.config.BaseConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * Support loading an Instance of a service, optionally providing it with configuration obtained
  * from the Kroxylicious configuration file.
  *
  * @param <T> the service type
+ * @param <S> the context type
  */
 public interface Contributor<T, S extends Context> {
```

Comment:
Note that Contributor still exists in this change, along with ContributorManager. I tried to keep the changes smaller.

One idea was to also remove these classes and move away from the idea of generically handling multiple types of contributors. This might reduce some of the complex generic code at the cost of duplicate boilerplate in Kroxylicious to get an instance for a typeName. Is it worth coming back to that in another PR?

Reply:
we've arriving at solution that allows a single class file to implement a filter - I think this might be sufficiently simple.